lang/java/compiler/src/main/java/org/apache/avro/compiler/specific/SpecificCompiler.java

```diff
@@ -328,6 +328,11 @@ private void getClassNamesOfPrimitiveFields(Schema schema, Set<String> result, S
       break;
     case ENUM:
     case FIXED:
```

Comment:
The `ENUM` case will fall through here!  This probably won't be an issue, but it would probably be more clear to give it its own `break;`

Reply:
Thanks for the reply @RyanSkraba .

This was actually done on purpose (ie. Enum and Fixed to check for logical types) as I initially thought to write a test using a custom enum conversion and logical type, that failed, but I feel is made possible and tested through https://github.com/apache/avro/pull/885/files#diff-a675c373b970779aafefc7cb7adf572c (where that PR is also based on this work).

However, if Enum should not actually be used in that way, will definitely add a break here.

Reply:
If that's the case, do you think it would be possible to unifying the logic with `result.add(javaType(scheme, true))` for all non-complex types?  I'm not sure what the real answer is, I'm interested in your opinion!

Reply:
Have done some testing, and unifying with `result.add(javaType(scheme, true))` appears perfectly logical. This method is only being used currently to determine the types that have conversions, and this will pick it up.

It performs a bit of unnecessary work if no logical type for the fixed/enum, as goes through the mangle path, but at the same time, it probably makes the code more consistent and potentially easier to follow imo.